minetest/minetest
# Settings GUI: Move shadows dropdown to Shaders, remove Most Used

Part of #13476

The aim of the new settings GUI is to make it much easier for users to find and change settings based on a logical categorisation of settings. Most Used was only intended to be temporary, after this PR and #13704 it's no longer needed

The shadows dropdown is now on the Shaders page. The dropdown index is no longer persisted, it's instead detected based on the actual settings. Changing shadow settings manually shows "Custom" on the dropdown.

Note that I intend to hide the dynamic shadow settings (except this dropdown) as advanced, I've kept the settings as-is so that it's easier to test the behaviour of the dropdown - especially "Custom".

![image](https://github.com/minetest/minetest/assets/2122943/3ca0bf46-0a68-4b73-b8d6-33b22a73161b)


## To do

This PR is Ready for Review

## How to test

* Clear your minetest.conf
* Open settings GUI, go to Shaders
* Check the dynamic shadows check box, see that the drop says Medium
* Set dynamic shadows to a preset, see the values below change
* Change a value below, see the drop down set to "Custom"
* Set a preset, see the values change again. Check that custom is not available in the drop down
